weighted_exclusivity_test/io.py

```diff
@@ -79,4 +83,37 @@ def output_table(args, setToPval, setToRuntime, setToFDR, setToObs, json_format=
                           setToRuntime=convert_dict_for_json(setToRuntime))
             json.dump( output, OUT )
 
-    
+# Output MCMC
+def output_mcmc(args, setsToFreq, setToPval, setToObs):
+    if args.json_format:
+        params = vars(args)
+        output = dict(params=params, setToPval=convert_dict_for_json(setToPval),
+                      setToObs=convert_dict_for_json(setToObs),
+                      setsToFreq=dict( (' '.join([ ','.join(sorted(M)) for M in sets ]), freq) for sets, freq in setsToFreq.iteritems() ))
+        with open(args.output_prefix + '.json', 'w') as OUT:
+            json.dump( output, OUT )
+    else:
+        # Output a gene set file
+        with open(args.output_prefix + '-sampled-collections.tsv', 'w') as OUT:
+            rows = []
+            for sets, freq in setsToFreq.iteritems():
+                row = [ ' '.join([ ','.join(M) for M in sets ]), freq ]
+                row.append( sum( -np.log10(setToPval[M] ** args.alpha) for M in sets ))
```

Comment:
Replace with `row.append( -args.alpha * sum( np.log10(setToPval[M]) for M in sets ))`.